### DISCUSSION TITLE
Trying to trigger a rule when POWER1 is already OFF and receives a new OFF from Alexa/hue emulation.

### DISCUSSION THREAD

**Discussion by Phlavio-SPR** [arendst/Tasmota#22085] [Q&A]
Having a Tasmotized IR Bridge, its easy to react an Alexa command to send some IR code.
Thing is I couldn't find a way to trigger a rule when receiving the same state that is current on POWER1.
![Alexa](https://github.com/user-attachments/assets/18f144ef-2a91-45db-bd2a-d5d94f374d0c)
In the example above, we have the reaction of three commands received from Alexa, all of them setting POWER1 to OFF.
I need to send an IR "OFF" command every time it receives an OFF from Alexa and an IR "ON" command every time it receives an ON from Alexa, regardless of the current state of POWER1.
As far I could see, only an actual POWER state change can trigger a rule.
Using SetOption59 and tele-Power isn't an option since it makes impossible to use more than one power state.
Did I miss something?
I appreciate any help. Thanks!

**Comment by sfromis** [comment]
Rule triggers with `Power1#state` are indeed only when the state is changed, as explicitly described in the docs. This is not directly related to the the incoming commands, but instead about changed power state for any reason, not limited to commands.

Not sure why you think that your approach with `SetOption59` and `tele-POWER` could not work for both power states, as the value should be available as `%value%` or triggers like `tele-POWER=OFF`

A more complicate technique: In a setup with MQTT, a build with `#define SUPPORT_MQTT_EVENT`, and `SetOption4` enabled, it seems to work to subscribe to the status messages about current power state whether it was changed or not.
> `subscribe power1,stat/mydevice/POWER,POWER`

And then rules like `on event#power1=OFF do ... endon`. Of course, it still would trigger if the power state is queried for other reasons, but does not react to the telemetry period messages.
